src/Processor/UpdatedOrderProcessor.php

```diff
@@ -0,0 +1,37 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Setono\SyliusOrderEditPlugin\Processor;
+
+use Doctrine\ORM\EntityManagerInterface;
+use Setono\SyliusOrderEditPlugin\Exception\NewOrderWrongTotalException;
+use Sylius\Component\Core\Inventory\Operator\OrderInventoryOperatorInterface;
+use Sylius\Component\Core\Model\OrderInterface;
+use Sylius\Component\Order\Processor\OrderProcessorInterface;
+
+final class UpdatedOrderProcessor
+{
+    public function __construct(
+        private OrderProcessorInterface $orderProcessor,
+        private OrderInventoryOperatorInterface $orderInventoryOperator,
+        private OrderProcessorInterface $afterCheckoutOrderPaymentProcessor,
+        private EntityManagerInterface $entityManager,
+    ) {
+    }
+
+    public function process(int $initialTotal, OrderInterface $newOrder): void
+    {
+        $newOrder->setState(OrderInterface::STATE_CART);
+        $this->orderProcessor->process($newOrder);
+        $this->afterCheckoutOrderPaymentProcessor->process($newOrder);
+        $newOrder->setState(OrderInterface::STATE_NEW);
+        $this->orderInventoryOperator->hold($newOrder);
+
+        if ($initialTotal < $newOrder->getTotal()) {
+            throw NewOrderWrongTotalException::occur();
+        }
+
+        $this->entityManager->flush();
```

Comment:
I think these lines belong in the controller or somewhere else. If they were removed, you could also remove the `$initialTotal` from the signature which I think would also make more sense

Reply:
👍 I can make a validator that would check the `$initalTotal < $newOrderTotal` condition and leave the entityManager in a controller 👌

Reply:
Yes, that check probably belongs in a validator 👍